pkg/cmd/webhooks/forward.go

```diff
@@ -0,0 +1,225 @@
+		IO:         f.IOStreams,
+		Config:     f.Config,
+	}
+	cmd := cobra.Command{
+		Use:   "forward --events=<event_types> --repo=<repo> --port=<port> [--host=<host>]",
+		Short: "Receive test webhooks locally",
+		Example: heredoc.Doc(`
+			# create a dev webhook for the 'issue_open' event in the monalisa/smile repo in GitHub running locally, and
+			# forward payloads for the triggered event to localhost:9999
+
+			$ gh webhooks forward --events=issue_open --repo=monalisa/smile --port=9999 --host=api.github.localhost
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if opts.EventTypes == nil {
+				return cmdutil.FlagErrorf("`--events` flag required")
+			}
+			if opts.Repo == "" {
+				return cmdutil.FlagErrorf("`--repo` flag required")
+			}
+			if opts.Host == "" {
+				fmt.Fprintf(opts.IO.Out, "No --host specified, connecting to default GitHub.com \n")
+				opts.Host = gitHubAPIProdURL
+			}
+
+			if opts.Port == 0 {
+				fmt.Fprintf(opts.IO.Out, "No --port specified, printing webhook payloads to stdout \n")
+			}
+
+			if runF != nil {
+				return runF(&opts)
+			}
+
+			wsURLString, err := createHook(&opts)
+			if err != nil {
+				return err
+			}
+
+			wsURL, err := url.Parse(wsURLString)
+			if err != nil {
+				return err
+			}
+
+			cfg, err := opts.Config()
+			if err != nil {
+				return err
+			}
+			token, _ := cfg.AuthToken(opts.Host)
+
+			for {
+				if retries >= maxRetries {
+					fmt.Fprintf(opts.IO.Out, "Unable to connect to webhooks server, forwarding stopped.\n")
+					return nil
+				}
+
+				err := handleWebsocket(&opts, token, wsURL)
+				if err != nil {
+					unwrapped := errors.Unwrap(err)
+					var syscallErr *os.SyscallError
+					// If the error is a TCP handleWebsocket error or a server disconnect (1006), retry connecting
+					if errors.As(unwrapped, &syscallErr) && (unwrapped.Error() == "connect: connection refused") || websocket.IsCloseError(err, websocket.CloseAbnormalClosure) {
+						retries += 1
+						time.Sleep(5 * time.Second)
+						continue
+					}
+					if websocket.IsCloseError(err, websocket.CloseNormalClosure) {
+						return nil
+					}
+				}
+			}
+		},
+	}
+	cmd.Flags().StringSliceVarP(&opts.EventTypes, "events", "E", []string{}, "Name of the event types to forward")
+	cmd.Flags().StringVarP(&opts.Repo, "repo", "R", "", "Name of the repo where the webhook is installed")
+	cmd.Flags().IntVarP(&opts.Port, "port", "P", 0, "Local port to receive webhooks on")
+	cmd.Flags().StringVarP(&opts.Host, "host", "H", "", "Host address of GitHub API (default: api.github.com)")
+	return &cmd
+}
+
+// createHook issues a request against the GitHub API to create a dev webhook
+func createHook(o *hookOptions) (string, error) {
+	httpClient, err := o.HttpClient()
+	if err != nil {
+		return "", err
+	}
+	apiClient := api.NewClientFromHTTP(httpClient)
+	path := fmt.Sprintf("repos/%s/hooks", o.Repo)
+	req := createHookRequest{
+		Name:   "cli",
+		Events: o.EventTypes,
+		Active: true,
```

Comment:
On a separate note: this might actually solve the "ping event race condition" if we set it to "false" then update it to true _after_ connecting to the websocket? Not sure if that's how it works: cc @DevinRiley

Reply:
Unfortunately that's not how it works, we still ping inactive hooks on create and I don't believe we send another ping when going from inactive -> active